### PR TITLE
open /dev/urandom when entropy context is initialized and keep open

### PR DESCRIPTION
**Pull request by jkarneges**
Proposed patch for #322. The user simply calls `entropy_init` before entering chroot.

### CODE REVIEW

**Comment by miklis** [comment]
Automatic CI verification build not done, please verify manually.

**Comment by mpg** [comment]
At first glance, looks good to me.

In order to accept your contribution, we'll request that you sign our contributor license agreement. Can you please get in touch with @sbutcher-arm or @pjbakker regarding the details? Thanks!

**Comment by jkarneges** [comment]
Happy to do the CLA but I'm not sure how to get in touch (neither Sam nor Paul have their email addresses public). Either of them can contact me at the address on my profile. Thanks.

**Comment by mpg** [comment]
I just sent an email to the three of you.

File: library/entropy_poll.c [hunk 1]
```diff
@@ -135,10 +140,27 @@ static int has_getrandom = -1;
 
 #include <stdio.h>
 
+#if !defined(HAVE_GETRANDOM)
+static FILE *urandom_file = NULL;
+#endif /* !HAVE_GETRANDOM */
+
+int mbedtls_platform_entropy_init( void )
+{
+#if !defined(HAVE_GETRANDOM)
```
Comment (gilles-peskine-arm) [inline library/entropy_poll.c]: `HAVE_GETRANDOM` doesn't guarantee that `getrandom` is available at runtime (we support older Linux kernel versions that some embedded devices run), so the `/dev/urandom` code needs to be included regardless.

**Comment by gilles-peskine-arm** [comment]
@jkarneges Thank you for your contribution! I've made a new pull request #1545 which builds on your work, fixes the case of getrandom() available at build time but not at runtime, and makes a few related improvements.

**Comment by sbutcher-arm** [comment]
This PR hasn't been active in a long time, and development work has been pursued in PR #1545. I am therefore closing this PR, and further work can happen there.

@jkarneges - if you'd like to pick this up again, please raise a new PR or push to the other PR #1545.

**Pull request closed without merging**